source/game_sa/Hud.cpp

```diff
@@ -342,9 +365,24 @@ void CHud::SetHelpMessageWithNumber(const char* text, int32 number, bool quickMe
     m_fHelpMessageStatUpdateValue = 0.0f;
 }
 
+// TODO Bad Loop
 // 0x588F60
 void CHud::SetMessage(char* text) {
-    plugin::Call<0x588F60, char*>(text);
+    // plugin::Call<0x588F60, char*>(text);
+    if (text) {
+        *m_Message = *text;
```

Comment:
Use

`memcpy(&m_Message, text, sizeof(char) * std::min(static_cast<int>(strlen(text)), 400));`

Reply:
Hm, why not `strcpy_s`?

Reply:
what about dis?
```suggestion
        if (text && *text) {
            strncpy(m_Message, text, 400u);
            m_Message[strlen(text)] = '\0';
        } else {
            m_Message[0] = '\0';
        }
```

Reply:
Okay, if text is to be truncated `strncpy_s`:
```suggestion
        if (text && *text) {
            strncpy_s(m_Message,  text, sizeof(m_Message) - 1); // Not sure if 400 is == sizeof(m_Message)
        } else {
            m_Message[0] = '\0';
        }
```

Reply:
Eng:

Your implementation is better)

Russia (Native):

Ваша реализация лучше)

Reply:
> what about dis?

in 376 line use 400u (but maximum index 399)